posts/process-csvs-from-amazon-s3-using-apache-flink-jhipster-and-kubernetes.md

```diff
@@ -0,0 +1,187 @@
+- [COVID-19 Vaccine Distribution Allocations by Jurisdiction - Moderna](https://data.cdc.gov/Vaccinations/COVID-19-Vaccine-Distribution-Allocations-by-Juris/b7pe-5nws)
+- [COVID-19 Vaccine Distribution Allocations by Jurisdiction - Pfizer](https://data.cdc.gov/Vaccinations/COVID-19-Vaccine-Distribution-Allocations-by-Juris/saz5-9hgg)
+
+The method `run` below is our "main" function:
+
+```java
+public static List<VaccineStatsVM> run() throws Exception {
+    // Connect to remote Flink and provide the Jar to be shipped
+    ExecutionEnvironment env = ExecutionEnvironment.createRemoteEnvironment("localhost", 8081, "target/jhipster-0.0.1-SNAPSHOT.jar.original");
+
+    // Read Moderna data
+    DataSource<VaccinePojo> moderna = env
+        .readCsvFile("s3a://tlebrun-playground/csv/COVID-19_Vaccine_Moderna.csv")
+        .ignoreFirstLine()
+        .parseQuotedStrings('"')
+        .includeFields("1000001") // Fields "Jurisdiction" and "Doses allocated for distribution week of 01/04"
+        .pojoType(VaccinePojo.class, "state", "allocatedDosesModerna");
+
+    // Read Pfizer data
+    DataSource<VaccinePojo> pfizer = env
+        .readCsvFile("s3a://tlebrun-playground/csv/COVID-19_Vaccine_Pfizer.csv")
+        .ignoreFirstLine()
+        .parseQuotedStrings('"')
+        .includeFields("100000001") // Fields "Jurisdiction" and "Doses allocated for distribution week of 01/04"
+        .pojoType(VaccinePojo.class, "state", "allocatedDosesPfizer");
+
+    // Read stats by state data
+    DataSource<StatsByStatePojo> statsByStates = env
+        .readCsvFile("s3a://tlebrun-playground/csv/COVID-19_Cases_and_Deaths_by_State.csv")
+        .ignoreFirstLine()
+        .parseQuotedStrings('"')
+        .includeFields("11100001")
+        .pojoType(StatsByStatePojo.class, "date", "state", "cases", "death");
+
+    // Filter stats on 01/04/2021
+    FilterOperator<StatsByStatePojo> dayStatsByStates = statsByStates
+        .filter(statsByState -> "01/04/2021".equals(statsByState.getDate()));
+
+    // Join the vaccine data and cases using state as key
+    // Sort by number of death
+    return moderna
+        .join(pfizer)
+        .where("state")
+        .equalTo("state")
+        .map(new MapToVaccineFunction())
+        .join(dayStatsByStates)
+        .where(new IdSelectorStateCode())
+        .equalTo("state")
+        .map(new MapToCovidFunction())
+        .sortPartition(new IdSelectorDeath(), Order.DESCENDING)
+        .collect();
+}
+```
+
+Reading a CSV is pretty simple and Flink will do all the parsing job for you. Manipulating each `DataSource` is also pretty easy and operations like filtering/mapping can done directly on the data. The COVID-19 cases/deaths data source is filtered to only keep the values for the date `01/04/2021` since we are using the vaccine data for this date too.
```

Comment:
```suggestion
Reading a CSV is pretty simple and Flink will do all the parsing for you. Manipulating each `DataSource` is also pretty easy and operations like filtering/mapping can done directly on the data. The COVID-19 cases/deaths data source is filtered to only keep the values for the date `01/04/2021` since we are using the vaccine data for this date too.
```